Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,185 @@
+/*global define*/
+define([
+        '../ThirdParty/Uri',
+        '../ThirdParty/when',
+        './defaultValue',
+        './defined',
+        './defineProperties',
+        './DeveloperError'
+    ], function(
+        Uri,
+        when,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError
+    ) {
+    "use strict";
+
+    var activeRequestsByServer = {};
+    var activeRequests = 0;
+
+    /**
+     * Because browsers throttle the number of parallel requests allowed to each server
+     * and across all servers, this class tracks the number of active requests in progress
+     * and prioritizes incoming requests.
+     *
+     * @exports RequestScheduler
+     *
+     * @see {@link http://wiki.commonjs.org/wiki/Promises/A|CommonJS Promises/A}
+     */
+    var RequestScheduler = function() {
```

Comment:
Let's mark this `@private` until we know how it is exposed, e.g., 3D Tiles will just use it under the hood, but if a user creates a `Model` instance for a glTF asset, they may choose to make the request immediately or put it in the scheduler.  At least, I'm pretty sure that is how we will want to do it.